src/core/ext/transport/cronet/transport/cronet_transport.c

```diff
@@ -51,612 +51,918 @@
 
 #define GRPC_HEADER_SIZE_IN_BYTES 5
 
-// Global flag that gets set with GRPC_TRACE env variable
+#define CRONET_LOG(...)                          \
+  {                                              \
+    if (grpc_cronet_trace) gpr_log(__VA_ARGS__); \
+  }
+
+/* TODO (makdharma): Hook up into the wider tracing mechanism */
 int grpc_cronet_trace = 1;
 
-// Cronet transport object
+enum OP_RESULT {
+  ACTION_TAKEN_WITH_CALLBACK,
+  ACTION_TAKEN_NO_CALLBACK,
+  NO_ACTION_POSSIBLE
+};
+
+/* Used for printing debug */
+const char *op_result_string[] = {"ACTION_TAKEN_WITH_CALLBACK",
+                                  "ACTION_TAKEN_NO_CALLBACK",
+                                  "NO_ACTION_POSSIBLE"};
+
+enum OP_ID {
+  OP_SEND_INITIAL_METADATA = 0,
+  OP_SEND_MESSAGE,
+  OP_SEND_TRAILING_METADATA,
+  OP_RECV_MESSAGE,
+  OP_RECV_INITIAL_METADATA,
+  OP_RECV_TRAILING_METADATA,
+  OP_CANCEL_ERROR,
+  OP_ON_COMPLETE,
+  OP_FAILED,
+  OP_SUCCEEDED,
+  OP_CANCELED,
+  OP_RECV_MESSAGE_AND_ON_COMPLETE,
+  OP_READ_REQ_MADE,
+  OP_NUM_OPS
+};
+
+const char *op_id_string[] = {"OP_SEND_INITIAL_METADATA",
+                              "OP_SEND_MESSAGE",
+                              "OP_SEND_TRAILING_METADATA",
+                              "OP_RECV_MESSAGE",
+                              "OP_RECV_INITIAL_METADATA",
+                              "OP_RECV_TRAILING_METADATA",
+                              "OP_CANCEL_ERROR",
+                              "OP_ON_COMPLETE",
+                              "OP_FAILED",
+                              "OP_SUCCEEDED",
+                              "OP_CANCELED",
+                              "OP_RECV_MESSAGE_AND_ON_COMPLETE",
+                              "OP_READ_REQ_MADE",
+                              "OP_NUM_OPS"};
+
+/* Cronet callbacks */
+
+static void on_request_headers_sent(cronet_bidirectional_stream *);
+static void on_response_headers_received(
+    cronet_bidirectional_stream *,
+    const cronet_bidirectional_stream_header_array *, const char *);
+static void on_write_completed(cronet_bidirectional_stream *, const char *);
+static void on_read_completed(cronet_bidirectional_stream *, char *, int);
+static void on_response_trailers_received(
+    cronet_bidirectional_stream *,
+    const cronet_bidirectional_stream_header_array *);
+static void on_succeeded(cronet_bidirectional_stream *);
```

Comment:
on what succeeded? Either change the name to something more descriptive or add docstrings (probably the former is best, docs go out of date quickly and compiler doesn't help...)